app/src/main/java/com/example/tasky/presentation/Navigation/Navigation.kt

```diff
@@ -18,7 +24,19 @@ fun Navigation(navController: NavHostController) {
             LoginScreen(navController = navController)
         }
         composable(Screen.AgendaScreen.route) {
+            val viewModel = hiltViewModel<AgendaViewModel>()
+            val state by viewModel.state.collectAsStateWithLifecycle()
+            val logoutEventsChannelFlow = viewModel.logoutResult
+            viewModel.dateDialogState = rememberMaterialDialogState()
 
+            AgendaScreen(
+                navController = navController,
```

Comment:
I'd recommend to keep composable screens free of NavController references. This has multiple reasons:
1. It creates a tight coupling. This means your UI components are not only responsible for displaying content but also for knowing how to navigate. This goes against the principle of separation of concerns, making your components less modular and harder to reuse.
2. With navigation logic embedded in your composables, testing these components in isolation becomes more challenging. You need to mock the navigation controller or setup navigation in your tests, which can complicate the testing process.
3. When navigation is controlled from multiple points in the app, it becomes more challenging to understand and manage the overall navigation flow. This can lead to bugs and difficulties in maintaining the code, especially as the complexity of the app grows.

Instead, I'd just perform the navigation either in the navigation `composable` block directly or creating a separate composable for that which just wraps the screen composable.